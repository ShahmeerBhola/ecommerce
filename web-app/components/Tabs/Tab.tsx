import React from 'react';

type Props = {
  selected: boolean;
  onClick: () => void;
};

const Tab: React.FunctionComponent<Props> = ({ children, selected, onClick }) => (
  <div
    className={`text-white font-bold py-2 px-4 uppercase cursor-pointer ${selected ? 'bg-red-100' : ''}`}
    onClick={onClick}
  >
    {children}
  </div>
);

export default Tab;
