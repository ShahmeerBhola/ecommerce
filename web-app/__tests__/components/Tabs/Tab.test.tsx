import React from 'react';
import renderer from 'react-test-renderer';
import Tab from '../../../components/Tabs/Tab';

describe('Tab component', () => {
  const onClick = (): void => console.log('hello');

  test('it renders properly when selected is true', () => {
    const tree = renderer
      .create(
        <Tab selected onClick={onClick}>
          <p>Tab 1</p>
        </Tab>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when select is false', () => {
    const tree = renderer
      .create(
        <Tab selected={false} onClick={onClick}>
          <p>Tab 1</p>
        </Tab>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
