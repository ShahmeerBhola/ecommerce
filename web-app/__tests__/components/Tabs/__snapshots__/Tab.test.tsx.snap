// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component it renders properly when select is false 1`] = `
<div
  className="text-white font-bold py-2 px-4 uppercase cursor-pointer "
  onClick={[Function]}
>
  <p>
    Tab 1
  </p>
</div>
`;

exports[`Tab component it renders properly when selected is true 1`] = `
<div
  className="text-white font-bold py-2 px-4 uppercase cursor-pointer bg-red-100"
  onClick={[Function]}
>
  <p>
    Tab 1
  </p>
</div>
`;
